### PR TITLE
Fix setting title of renamed terminal windows

### LINKED ISSUE #99072: Include terminal name in terminal textarea title
Currently focusing the terminal reads something like "Terminal 1", "Terminal 2". It should read "Terminal 1, bash", "Terminal 2, (custom title)", etc.

Context: https://github.com/microsoft/vscode/issues/95573#issuecomment-637445449

### PR DESCRIPTION
**Pull request by plainerman**
This PR fixes #99072

As of now, when renaming terminal windows, the title of the terminal will not be updated. This makes it incompatible with text-to-speech programs. Whenever the title of a terminal changes, this change should propagate.

I have moved the required changes into the `setTitle` method of a terminal instance.

To try this fix, do the following:

* Call the action "Terminal: Rename" by pressing Ctrl + Shift + P and searching for it
* Give a new desired name
* Start the narrator or any other text-to-speech program
* Click the "Terminal" Tab
* It now reads the custom title

This change also ensures that the title is set in the beginning, meaning that the title is set accordingly for predefined names (e.g. "powershell").

### CODE REVIEW

**Comment by isidorn** [comment]
@meganrogge please let me know if you need help reviewing this as I was involved in the discussion on the issue

**Comment by plainerman** [comment]
@meganrogge I was wondering whether there is anything I can do to help you getting the PR merged?

Also, as this is my first contribution, I wanted to ask whether there is something wrong because the expected status of the one test has not been reported yet?

**Comment by isidorn** [comment]
@plainerman I believe @meganrogge was on vacation. She should respond to your questions soon.
As for the test, no worries about that, this is expected.

**Comment by plainerman** [comment]
@meganrogge, I hope you had a nice time off :)

I wanted to ask whether there are any updates regarding this PR? I'm eager to getting it merged, so don't hesitate to contact me if there is anything I can do to assist you.

**Comment by isidorn** [comment]
Maybe @Tyriar can help, he is also back from vacation.

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 1]
```diff
@@ -518,13 +518,6 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			throw new Error('xterm elements not set after open');
 		}
 
-		// Check if custom Terminal title exists and set same
-		if (this._title.length > 0) {
-			xterm.textarea.setAttribute('aria-label', nls.localize('terminalTextBoxAriaLabelNumberAndTitle', "Terminal {0}, {1}", this._id, this._title));
-		} else {
-			xterm.textarea.setAttribute('aria-label', nls.localize('terminalTextBoxAriaLabel', "Terminal {0}", this._id));
-		}
-
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:527]: Thanks for the fix! I think we also need to update the title here for the following case:

1. Terminal launches
2. Terminal title gets set (textarea didn't exist)
3. Attach is called

Can the code to update `aria-label` be factored out into a new function and called here also?

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts (hunk 1 above)
Comment (plainerman) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts]: I have put it in a private function and called it in `_attachToElement`. Is there anything left to do, to fulfill 1-3?

**Review by Tyriar — APPROVED** [review]
Perfect, thanks!

**Comment by plainerman** [comment]
Thank you for the review @Tyriar !

And also thank you @isidorn for your continued support in this PR. :)

**Pull request merged**